apps/coscrad-frontend/src/components/file-upload/file-upload-item.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+    Article as ArticleIcon,
+    AudioFile as AudioFileIcon,
+    Clear as ClearIcon,
+    Image as ImageIcon,
+    InsertDriveFile as InsertDriveFileIcon,
+    VideoCameraBack as VideoCameraBackIcon,
+} from '@mui/icons-material';
+import { Card, CardContent, Grid, IconButton, Tooltip, Typography } from '@mui/material';
+import LinearProgressWithLabel from '../linear-progress-with-label/linear-progress-with-label';
+import { FileUploadStatus, FileWithProgress } from './types/file-with-progress';
+
+const getFileIcon = (mimeType: string) => {
+    if (mimeType.startsWith('image/')) return ImageIcon;
+    if (mimeType.startsWith('video/')) return VideoCameraBackIcon;
+    if (mimeType.startsWith('audio/')) return AudioFileIcon;
+    if (mimeType === 'application/pdf') return ArticleIcon;
+    return InsertDriveFileIcon;
+};
+
+const formatFileSize = (bytes: number) => {
```

Comment:
We might want to move this to an `@coscrad/math` lib.

Reply:
I've added a comment about this possibility.

Reply:
[Here](https://coscrad.atlassian.net/browse/CWEBJIRA-298) is a story for a unit test.